Merge BRINGUP into main

minecraftchest1 opened the pull request.

minecraftchest1 committed:
Merge Wyatt into BRINGUP (#15)

* added a working mecanum drivesubsystem and control

* added motors for launcher mechanisum (kit bot)

* Add REVlib

* RevLib bringup

* Create odomotry object.

Add constants for relative wheel locations to Constants.java. Add
GetEncoders to MechanumDriveSubsystem to get encoders from
LimelightNavigation (this is subject to change). Get encoders from
MechanmDriveSubsytem and use them to create odometry objects on
LimelightNavigation.

* Merge in stashed changes

* Set robot pose based on limelight.

* Pass encoders into LimelightNavigation constructor.

* Troubleshoot perodic in subsystem.

* Make code work.

Add empty rotations, fix port forward.

* Add Pidgeon2 Can ID

* Use gyro angle in LimelightNavigation().

---------

Co-authored-by: dragonking1000 <zachariahjackson2018@outlook.com>
Co-authored-by: Wyatt Jackson <Wyatt Jackson@ad.acs-k12.us>

minecraftchest1 committed:
Merge Zach into BRINGUP (#16)

* added a working mecanum drivesubsystem and control

* added motors for launcher mechanisum (kit bot)

* RevLib bringup

* added intake subsystem and changed motor speed to be in Constants.java

* removed the joystick import

* added the REVLib.json libray and uppdated gitnore

* cahnged the intake to use a seperate motor than the launcher

* added a subsystem for the trapscoring mecanisum and added the motor id to constants.java

* added a score device for the trap and added all the controls and constant angles necisary

* Add PWMSparkMax channels to `Constants` class.

* Add .vscode to gitignore.

* Create MechaniumDrive subsystem.

* added a working mecanum drivesubsystem and control

* added motors for launcher mechanisum (kit bot)

* RevLib bringup

* added intake subsystem and changed motor speed to be in Constants.java

* removed the joystick import

* added the REVLib.json libray and uppdated gitnore

* cahnged the intake to use a seperate motor than the launcher

* added a subsystem for the trapscoring mecanisum and added the motor id to constants.java

* added a score device for the trap and added all the controls and constant angles necisary

* Create MechaniumDrive subsystem.

* added a working mecanum drivesubsystem and control

* RevLib bringup

* added a subsystem for the trapscoring mecanisum and added the motor id to constants.java

* Create MechaniumDrive subsystem.

* added a working mecanum drivesubsystem and control

* RevLib bringup

* added a subsystem for the trapscoring mecanisum and added the motor id to constants.java

* removed the example command and subsystem

* added the REVPhysicsSim stuff

* seems i forgot to add this file to the last commit

* fixed REVPhysicsSim in IntakeMechanisum.java and added new CANSparkMAxId in Constants.java

* removed unused import in intake

* changed REVPhysicsSim activation to when the robot is in simulation

* removed unused imports

* added some code

* added speed tracking for the mecanum drive base

* added an output to smart dash board for motor speeds

* moved back to non simulation periodic

* removed the simulation peridic in launch mech subsystem

* added code to test periodic function

* updates some motor names and tried to debug periodic problem.

---------

Co-authored-by: dragonking1000 <zachariahjackson2018@outlook.com>

minecraftchest1 committed:
uncomited out robotcontainer.java

minecraftchest1 committed:
Merge pull request #18 from frc538/Zach

Uncomment use of RobotContainer

minecraftchest1 committed:
Format code and add missing }

Indent Constants.java using tabs.

minecraftchest1 committed:
Add CTRE Phoenix vendor library.

minecraftchest1 committed:
Capitolize refrences to Constants.Misc

During merging of Zach and Wyatt into bringup, Constants.Misc was the name used for misclanous items. This commit capitolizes all refrences to the lowercase version.

minecraftchest1 committed:
Remove Example imports

Remove imports for ExampleCommand and ExampleSubsystem.

minecraftchest1 committed:
Fix spelling of RearRightriveMotor

minecraftchest1 committed:
Remove unit test

drewwhis commented on .gitignore:
```diff
@@ -176,3 +176,8 @@ logs/
 # Folder that has CTRE Phoenix Sim device config storage
 ctre_sim/
+.vscode/launch.json
+.vscode/settings.json
```
these should not be ignored - they include the debug settings and Java config for the project

drewwhis commented on .gitignore:
```diff
@@ -176,3 +176,8 @@ logs/
+.vscode/settings.json
+networktables.json
+simgui-ds.json
+simgui.json
```
you _can_ ignore these, but there's not strictly an advantage to do so, unless you have a specific reasoning in mind.

drewwhis commented on LimeLightTest/.Glass/glass.json:
```diff
@@ -0,0 +1,44 @@
+{
```
The `LimeLightTest` folder/project should be removed - it looks like it was meant to be an independent test project that got put into the main project.

drewwhis commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -17,21 +19,52 @@
+		public static final double ampLaunchSpeed = 0.5;
+		public static final int startAngle = 0;
+		public static final int loadAngle = 90;
+		public static final int dropAngle = 180;
```
you _may_ want to save these off as `Rotation2d` instead of `int`, or you may want to add the word `degrees` into the variable name to avoid confusion (if you leave them as `int`)

drewwhis commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -23,14 +37,18 @@
+
+  private Map<String, RelativeEncoder> Encoders = m_Drive.GetEncoders();
+
+  private final LimelightNavigation m_Navigation = new LimelightNavigation(Encoders);
```
I think you can leave limelight navigation as its own subsystem, but it _may_ be better or cleaner to just include the navigation components all as part of the Drive Subsystem - if the other subsystems are taking actions based off the navigation, then it's just a matter of writing triggers.

minecraftchest1 replied:
Moved into #26

drewwhis commented on src/main/java/frc/robot/commands/Autos.java:
```diff
@@ -3,16 +3,10 @@
-import frc.robot.subsystems.ExampleSubsystem;
-import edu.wpi.first.wpilibj2.command.Command;
-import edu.wpi.first.wpilibj2.command.Commands;
+//import edu.wpi.first.wpilibj2.command.Command;
+//import edu.wpi.first.wpilibj2.command.Commands;
```
in general, delete code - don't comment it out. `git` maintains a revision history for you, so commented code just clutters up the files and makes them less readable. there may be other places where this is happening, so double check the other files.

drewwhis commented on src/main/java/frc/robot/subsystems/IntakeMechanisum.java:
```diff
@@ -0,0 +1,39 @@
+    float stallTorque = Constants.Misc.stallTorque;
+    float freeSpeed = Constants.Misc.freeSpeed;
+
+    REVPhysicsSim.getInstance().addSparkMax(intake, stallTorque, freeSpeed);
```
instead of this version of `addSparkMax`, you can also use `addSparkMax(motor, DCMotor.getNEO(1));`, and it will populate all of the NEO parameters for you. `DCMotor` also has methods for other common motors.

drewwhis commented on src/main/java/frc/robot/subsystems/LanuchMechanisumSubsystem.java:
```diff
@@ -4,24 +4,55 @@
+      float stallTorque = Constants.Misc.stallTorque;
+      float freeSpeed = Constants.Misc.freeSpeed;
 
+      REVPhysicsSim.getInstance().addSparkMax(top, stallTorque, freeSpeed);
+      REVPhysicsSim.getInstance().addSparkMax(bottom, stallTorque, freeSpeed);
+      REVPhysicsSim.getInstance().addSparkMax(staging, stallTorque, freeSpeed);
+      REVPhysicsSim.getInstance().addSparkMax(pickup, stallTorque, freeSpeed);
```
see the comment on `IntakeMechanisum`

drewwhis commented on src/main/java/frc/robot/subsystems/MecanumDriveSubsystem.java:
```diff
@@ -43,14 +56,57 @@ public MecanumDriveSubsystem() {
+    float stallTorque = Constants.Misc.stallTorque;
+    float freeSpeed = Constants.Misc.freeSpeed;
+    if (RobotBase.isSimulation()) {
+      REVPhysicsSim.getInstance().addSparkMax(frontLeft, stallTorque, freeSpeed);
+    REVPhysicsSim.getInstance().addSparkMax(frontRight, stallTorque, freeSpeed);
+    REVPhysicsSim.getInstance().addSparkMax(rearLeft, stallTorque, freeSpeed);
+    REVPhysicsSim.getInstance().addSparkMax(rearRight, stallTorque, freeSpeed);
+
```
see the comment on `IntakeMechanisum`

drewwhis commented on src/test/java/frc/robot/subsystems/MecanumTest.java:
```diff
@@ -1,33 +0,0 @@
-// Copyright (c) FIRST and other WPILib contributors.
```
don't delete this file - it's used to demonstrate the unit testing functionality. if the build fails, then this file needs to be modified

minecraftchest1 committed:
Remove LimeLightTest project.

minecraftchest1 committed:
Remove unneded ignores.

dragonruler1000 committed:
fixed a problem with a int name

minecraftchest1 committed:
Add .vscode

dragonruler1000 committed:
added Degrees sufix to the constants names for the angles

dragonruler1000 committed:
Switctced the REVPhysicsSim freespeed and freetork to DCMotor.getNEO(1));

dragonruler1000 committed:
Merge branch 'BRINGUP' into Zach

dragonruler1000 committed:
Merge pull request #25 from frc538/Zach

Fixed some comments

minecraftchest1 committed:
Remove code outside of class.

minecraftchest1 committed:
Fix missing symbol

minecraftchest1 committed:
Uncomment imports in autos.

minecraftchest1 commented on src/main/java/frc/robot/commands/Autos.java:
Uncommented; it looked like template code that should not be deleted.

minecraftchest1 reviewed (commented):
Everything looks good.

npatrick04 commented on src/main/java/frc/robot/subsystems/MecanumDriveSubsystem.java:
```diff
@@ -43,14 +54,55 @@ public MecanumDriveSubsystem() {
   public void periodic() {
-    // This method will be called once per scheduler run
+  // This method will be called once per scheduler run
+    // double frontLeftSpeed = frontLeft.getEncoder().getVelocity();
```
Delete commented out code, or use it.  It wouldn't be bad for the periodic function to post drive speeds to SmartDashboard.

dragonruler1000 replied:
That I left in there while I tested the periodic code so I could figure out where the problem I was seeing was coming from.

drewwhis commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -17,21 +17,52 @@
+    	public static final double intSpeed = .25;
+		
+    	// Positions of the Drive Wheels relative to the center of the Robot.
+    	public static final double FrontLeftDriveWheel_Position_X = -0.3;
+    	public static final double FrontLeftDriveWheel_Position_Y = 0.3;
+    	public static final double FrontRightDriveWheel_Position_X = 0.3;
+    	public static final double FrontRightDriveWheel_Position_Y = 0.3;
+    	public static final double RearLeftDriveWheel_Position_X = -0.3;
+    	public static final double RearLeftDriveWheel_Position_Y = -0.3;
+    	public static final double RearRightDriveWheel_Position_X = 0.3;
+    	public static final double RearRightDriveWheel_Position_Y = -0.3;
```
these names should have units on them (_e.g._, `FrontLeftDriveWheel_Position_X_meters` if the unit is meters)

minecraftchest1 replied:
#29

drewwhis commented on src/main/java/frc/robot/subsystems/IntakeMechanisum.java:
```diff
@@ -0,0 +1,36 @@
+
+    intake = new CANSparkMax(Constants.CANSparkMaxID.intake, MotorType.kBrushless);
+
+    REVPhysicsSim.getInstance().addSparkMax(intake, DCMotor.getNEO(1));
```
I think we should only add devices to the physics sim `if (RobotBase.isSimulation())`

minecraftchest1 replied:
#31

drewwhis commented on src/main/java/frc/robot/subsystems/IntakeMechanisum.java:
```diff
@@ -0,0 +1,36 @@
+  }
+
+
+  public void intake(){
+    intake.set(Constants.Misc.intakeSpeed);
+  }
```
you've got an `intake` method, but you may want to add an `eject` method (to spit out a piece that was mistakenly brought in, for example) and a `stop` method (in case you need more granular control). I wouldn't hold this up for that though - something to consider.

minecraftchest1 replied:
#30

drewwhis commented on src/main/java/frc/robot/subsystems/LanuchMechanisumSubsystem.java:
```diff
@@ -4,24 +4,54 @@
 
-  topLaunch = new PWMSparkMax(Constants.PWM.topLaunch);
-  bottomLaunch = new PWMSparkMax(Constants.PWM.bottomLaunch);
+    top = new CANSparkMax(Constants.CANSparkMaxID.topRight, MotorType.kBrushless);
+    bottom = new CANSparkMax(Constants.CANSparkMaxID.topLeft, MotorType.kBrushless);
+    staging = new CANSparkMax(Constants.CANSparkMaxID.staging, MotorType.kBrushless);
+    pickup = new CANSparkMax(Constants.CANSparkMaxID.launcherLoad, MotorType.kBrushless);
```
you may want to reconsider how you have your `Constants.CANSparkMaxID` values named, but I wouldn't consider it a blocker here.

minecraftchest1 replied:
#29

drewwhis commented on src/main/java/frc/robot/subsystems/LanuchMechanisumSubsystem.java:
```diff
@@ -4,24 +4,54 @@
+    staging.follow(pickup);
+
 
+      REVPhysicsSim.getInstance().addSparkMax(top, DCMotor.getNEO(1));
+      REVPhysicsSim.getInstance().addSparkMax(bottom, DCMotor.getNEO(1));
+      REVPhysicsSim.getInstance().addSparkMax(staging, DCMotor.getNEO(1));
+      REVPhysicsSim.getInstance().addSparkMax(pickup, DCMotor.getNEO(1));
```
need `if (RobotBase.isSimulation())`

minecraftchest1 replied:
#31

drewwhis commented on src/main/java/frc/robot/subsystems/LanuchMechanisumSubsystem.java:
```diff
@@ -4,24 +4,54 @@
+
+public void launchAmp(){
+  //launch into the amp
+  top.set(Constants.Misc.ampLaunchSpeed);
```
you probably want to set `pickup` to run at `0` speed with this method, since the mechanism should only be running one set of motors at a time, based on my reading

minecraftchest1 replied:
#33

drewwhis commented on src/main/java/frc/robot/subsystems/LanuchMechanisumSubsystem.java:
```diff
@@ -4,24 +4,54 @@
+}
+
+public void initaliseLauncher(){
+  pickup.set(Constants.Misc.intSpeed);
```
there may be a similar consideration here (as in `launchAmp`)

minecraftchest1 replied:
#33

drewwhis commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -59,20 +69,33 @@ private void configureBindings() {
-    // cancelling on release.
-    //m_driverController.b().whileTrue(m_exampleSubsystem.exampleMethodCommand());
-  }
+    contrJoystick.button(1).onFalse(Commands.run(() -> {
+      m_LaunchMech.launchSpeaker();
+    }, m_LaunchMech));
```
you may want to consider `onFalse` vs. `whileTrue` as well as `Commands.run` vs. `Commands.startEnd` - not necessarily that this isn't right - you may find better options with those when testing gets here. but don't let this block you.

minecraftchest1 replied:
#32

drewwhis commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -59,20 +69,33 @@ private void configureBindings() {
-    // An example command will be run in autonomous
-    return Autos.exampleAuto(m_exampleSubsystem);
-  }
+    contrJoystick.button(2).onFalse(Commands.run(() -> {
+      m_LaunchMech.launchAmp();
+    },m_LaunchMech));
+
+    contrJoystick.button(3).whileTrue(Commands.run(() -> {
+      m_Intakemech.intake();
+    },m_Intakemech));
+
+    contrJoystick.button(3).onFalse(Commands.run(() -> {
+      m_TrapScoreSubsystem.startAngle();
+    },m_TrapScoreSubsystem));
 
+    contrJoystick.button(4).onFalse(Commands.run(() -> {
+      m_TrapScoreSubsystem.loadAngle();
+    },m_TrapScoreSubsystem));
+
+    contrJoystick.button(5).onFalse(Commands.run(() -> {
+      m_TrapScoreSubsystem.dropAngle();
+    },m_TrapScoreSubsystem));
```
same as above:

- onFalse vs whileFalse
- onTrue vs whileTrue
- Commands.run vs Commands.startEnd

but don't let this block you.

minecraftchest1 replied:
#32

drewwhis reviewed (changes requested):
some of my comments aren't needs right now (and I tried to mark those as such). there should only be a couple of things that I think really _need_ to be changed right now.

minecraftchest1 committed:
Change wheel positions type and name.

Changed Relative Wheel Locations from `double` to `Translation2D`, and
added Meters suffix.

minecraftchest1 committed:
Remove unfinished statement.

drewwhis commented on src/main/java/frc/robot/subsystems/LimelightNavigation.java:
```diff
@@ -0,0 +1,87 @@
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+	if ( NetworkTableInstance.getDefault().getTable("limelight").getEntry("tv").getDouble(0) != 0 )
```
is there a reason we're pulling entries out of `NetworkTables` manually instead of using the Limelight Helpers? https://github.com/LimelightVision/limelightlib-wpijava/blob/main/LimelightHelpers.java (specifically `getLimelightNTTableEntry`)?

minecraftchest1 replied:
#19

drewwhis reviewed (dismissed):
primarily asking why we aren't using the LimelightHelpers (https://github.com/LimelightVision/limelightlib-wpijava/blob/main/LimelightHelpers.java)

we should be able to just add this as a file are the top level of the `robot` folder (same place as the `Constants.java` file - this would prevent us from having to write/chain verbose NetworkTables calls

dragonruler1000 committed:
some updates

dragonruler1000 committed:
Merge branch 'Zach' of https://github.com/frc538/2024-game into Zach

dragonruler1000 committed:
updated some code/did a git pull

minecraftchest1 committed:
Add LimelightHelpers class

minecraftchest1 committed:
Use LimelightHelpers class.

dragonruler1000 committed:
fixed issue 31, 33, and part of 29.

dragonruler1000 committed:
commented code to be deleted

minecraftchest1 committed:
Use LimelightHelpers.

minecraftchest1 committed:
Only create simulation objects if simulating.

Fixes #31

minecraftchest1 committed:
Merge Zach's code in.

minecraftchest1 commented:
> primarily asking why we aren't using the LimelightHelpers (https://github.com/LimelightVision/limelightlib-wpijava/blob/main/LimelightHelpers.java)
> 
> we should be able to just add this as a file are the top level of the `robot` folder (same place as the `Constants.java` file - this would prevent us from having to write/chain verbose NetworkTables calls

Nothing prevented this. I just forgot to use it. I am now using it and is cleaned the code up a lot.
